modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/OpenAPIUtil.java

```diff
@@ -143,7 +143,9 @@ private static String _getOperationScope(Operation operation) {
 			sb.append(",");
 		}
 
-		sb.setIndex(sb.index() - 1);
+		if (!parameters.isEmpty()) {
```

Comment:
Hi @tototrinh , according to our discussion please consider:
1. moving the checking condition to the top of the function. If parameters are empty, return an empty string for example StringPool.BLANK 
2. please double-check if we still need to add the empty string to the array list at line 59. Looks like it does not matter if the scope is empty here: https://github.com/Vy-CES/liferay-portal/blob/master/modules/apps/batch-planner/batch-planner-rest-impl/src/main/java/com/liferay/batch/planner/rest/internal/resource/v1_0/SiteScopeResourceImpl.java#L75

Reply:
Do we need to write or update the integration test for this method in OpenAPIUtilTest?

Reply:
Thanks @Vy-CES for your feedback. I updated the code, please help take a look again. Thanks.